lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -1937,8 +1969,17 @@ def _get_deployable_vhd_path(
         """
         matches = SAS_URL_PATTERN.match(vhd_path)
         if not matches:
-            return vhd_path
-        log.debug("found the vhd is a sas url, it may need to be copied.")
+            vhd_path = self._generate_sas_token(vhd_path, location, log)
```

Comment:
This should be split into two method. It doesn't need to check if it's sas token again. 1) Check if it's in the same location. If yes, return the vhd path. 2) Generate sas token, and go with next logics.

Reply:
updated